README.md

```diff
@@ -51,10 +51,12 @@ Note that the `*_path` elements are for very specific use cases where one might
 &nbsp;
 
 ## `restapi` datasource configuration
+Note that you must set the 'id_attribute' key on the restapi provider in order to use this datasource since it is used to set the ID of the discovered object that matches the search criteria.
 - `path` (string, required): The API path on top of the base URL set in the provider that represents objects of this type on the API server.
-- `search_key` (string, required): When reading search results from the API, this key is used to identify the specific record to read. This should be a unique record such as 'name'.
-- `search_value` (string, required): The value of 'search_key' will be compared to this value to determine if the correct object was found. Example: if 'search_key' is 'name' and 'search_value' is 'foo', the record in the array returned by the API with name=foo will be used.
-- `results_key` (string, required): When issuing a GET to the path, this JSON key is used to locate the results array. The format is 'field/field/field'. Example: 'results/values'. If omitted, it is assumed the results coming back are to be used exactly as-is.
+- `search_key` (string, required): When reading search results from the API, this key is used to identify the specific record to read. This should be a unique field within each record such as 'name' or a path to such an field in the form 'field/field/field'. Example: 'attributes/name' would search for the name field under the attributes map of each record found within the portion of the API response indicated by 'results_key'.
+- `search_value` (string, required): The value found in the last field in 'search_key' of each record will be compared to this value to determine if the correct record was found. Example: if 'search_key' is 'name' and 'search_value' is 'foo', the record in the array returned by the API with name=foo will be used.
+- `results_key` (string, required): When issuing a GET to the path, this JSON key is used to locate the results array. The format is 'field/field/field'. Example: 'results/values' will look for an array of records under the results/values portion of the API response. If omitted, it is assumed the results coming back are to be used exactly as-is.
+- `single_object` (boolean, optional): If set to true, indicates that the API call will return a single object instead of an array of objects. In this case, 'results_key', 'search_key', and 'search_value' are all ignored. We recommend setting them all to to "". 
```

Comment:
I don't doubt the use case here, but it seems like this may already covered when using `terraform import`? Is the difference here that a user may not want to bring the object under management (so won't use `import`), but would still want to be able to refer to fields in the object?

Reply:
I also thought about using import on the resource instead of using the datasource.

For my customer's vRO use case, they create an restapi_object with a path looking like `/<workflow_id>/executions` and then create a restapi_object datasource with path `/<workflow_id>/executions/id` where `id` is the ID returned for the resource.  With the datasource, then can run `terraform apply` multiple times and get updated information on the status of the workflow execution from the datasource.

In this case, importing would not be desired because the resource that would be imported has already been created by Terraform. 

However, it now occurs to me that they might be able to just run `terraform apply` multiple times and get updated status about the workflow execution from the resource itself using outputs that work against the resource's api_data attribute.  It occurs to me that running `terraform apply` a second time will cause Terraform to do a read against the read_path of the resource and that this would update the api_data attribute. Do you also think that would happen?  I think it certainly will happen if each `terraform apply` causes the resourceRestApiRead function to be called.

Reply:
I will do some testing on this.  If the read does get done for each apply, then we could drop the single_object attribute and associated code.

Reply:
So, I have done testing and have confirmed that each apply will re-read the current state of the object from the API server and that api_data will be updated.  So, in reality, my single_object syntax is not needed since there is no need to use a datasource if you have already created a resource that maps to the object the datasource would return.  You can just use outputs that refer to the api_data attribute of the resource.

I am going to remove the single_object code.

Reply:
I removed single_object attribute and related code and did new commit.

Reply:
OK, great - thanks. Sorry for not getting back sooner, but the read of every object *should* be done during the "refreshing state" phase of the apply.